src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -14,11 +14,13 @@
 
     Optional<User> findFirstByUsername(String userName);
 
-    @Query("SELECT u FROM User u WHERE "
+    @Query("SELECT u FROM User u WHERE ("
         + " (u.birthdate BETWEEN :dateStart AND :dateEnd)"
-        + " OR ( cast(:dateStart as date) IS NULL AND u.birthdate < :dateEnd )"
-        + " OR ( cast(:dateEnd as date) IS NULL AND u.birthdate > :dateStart )"
-        + " OR (u.name LIKE %:name%)")
+        + "     OR (( cast(:dateStart as date) IS NULL AND u.birthdate <= :dateEnd ))"
+        + "     OR (( cast(:dateEnd as date) IS NULL AND u.birthdate >= :dateStart ))"
+        + "     OR (( cast(:dateEnd as date) IS NULL AND cast(:dateStart as date) IS NULL ))"
+        + " )"
+        + " AND ( :name = '' OR lower(u.name) LIKE lower(concat('%', :name,'%')) )")
```

Comment:
Only thing you're missing is adding the default value of  '' to the name parameter in the search endpoint in the controller.

Reply:
Ok! Done, I added the required=false to every field on those controllers methods